hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileWriter.java

```diff
@@ -77,6 +81,8 @@ public HoodieHFileWriter(String instantTime, Path file, HoodieHFileConfig hfileC
     this.file = HoodieWrapperFileSystem.convertToHoodiePath(file, conf);
     this.fs = (HoodieWrapperFileSystem) this.file.getFileSystem(conf);
     this.hfileConfig = hfileConfig;
+    this.schema = schema;
+    this.schemaRecordKeyField = Option.ofNullable(schema.getField(HoodieMetadataPayload.SCHEMA_FIELD_ID_KEY));
```

Comment:
Can't this be passed from the metadata code path that actually instantiates the writer? Then we need not hardcode anything? What am I missing?

Reply:
i.e a new property `hoodie.hbase...key.field.name` or something, which is always set to `HoodieMetadataPayload.SCHEMA_FIELD_ID_KEY`

Reply:
here we do hfileConfig.getKeyFieldID or name or something?

Reply:
best to pass via hFileConfig with default as "key"

Reply:
It would be even simpler to migrate HoodieMetadataPayload.SCHEMA_FIELD_ID_KEY to HoodieHFileWriter. 

In other words, for now lets require HFile schemas to have a field called "key". In future, if the need arises for HFile as a real data file format for HUDI dataset outside of metadata table, we can customize this to allow configuring the key field.

Reply:
Incorporated vinoth's suggestion on using the storage config property and letting the HFileWriter use the config to let the writer know about the key field.